### PR TITLE
[DOCS] Revise use of the term "API"

### PR DESCRIPTION
**Pull request by klavavej**
Resolves https://greatexpectations.atlassian.net/browse/DOC-1133 according to the guidance linked in that issue

- [X] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [X] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB], [MINORBUMP]
- [ ] Code is linted - run `invoke lint` (uses `ruff format` + `ruff check`)
- [X] Appropriate tests and docs have been updated

For more information about contributing, visit our [community resources](https://docs.greatexpectations.io/docs/core/introduction/community_resources#contribute-code-or-documentation).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -57,10 +57,10 @@ They are listed in the order in which GX is prioritizing the support issues:
 ## Contribute
 We deeply value the contributions of our community. We're now accepting PRs for bug fixes.
 
-To ensure the long-term quality of the GX Core codebase, we're not yet ready to accept feature contributions to the parts of the codebase that don't have clear APIs for extensions. We're actively working to increase the surface area for contributions. Thank you for being a crucial part of GX Core!
+To ensure the long-term quality of the GX Core codebase, we're not yet ready to accept feature contributions to the parts of the codebase that don't have clear interfaces for extensions. We're actively working to increase the surface area for contributions. Thank you for being a crucial part of GX Core!
```
Comment (AkshatAtGX) [inline README.md:60]: I'm not sure I understand this sentence. What is a clear interface for extensions?

File: README.md [hunk 2]
```diff
@@ -57,10 +57,10 @@ They are listed in the order in which GX is prioritizing the support issues:
...
 
 ### Levels of contribution readiness
-🟢 Ready. Have a clear and public API for extensions.
+🟢 Ready. Have a clear and public interface for extensions.
```
Comment (AkshatAtGX) [inline README.md:63]: If I understand correctly, we're trying to communicate what parts of GX Core are ready for public contribution. Is that right?

If so, I'm not sure "interface" is the right word. But, I also don't have a suggestion at the moment. Need to think about it.

File: contrib/capitalone_dataprofiler_expectations/README.md [hunk 3]
```diff
@@ -26,7 +26,7 @@ print(json.dumps(readable_report, indent=4))
 ```
 Note: The Data Profiler comes with a pre-trained deep learning model, used to efficiently identify **sensitive data** (PII / NPI). If desired, it's easy to add new entities to the existing pre-trained model or insert an entire new pipeline for entity recognition.
 
-For API documentation, visit the [documentation page](https://capitalone.github.io/DataProfiler/).
+For more information, visit the [documentation page](https://capitalone.github.io/DataProfiler/).
```
Comment (AkshatAtGX) [inline contrib/capitalone_dataprofiler_expectations/README.md:29]: FWIW I think "API documentation" is fine to say, in this context, because it's referring to the API of the CapitalOne Data Profiler. But "information" also works, so this change is fine.

File: docs/docusaurus/docs/cloud/expectations/manage_expectations.md [hunk 4]
```diff
@@ -236,21 +236,21 @@ If your Data Asset has at least one DATE or DATETIME column, you can define a Ba
 
 In GX Cloud, Expectations can be GX-managed or API-managed.
 - GX-managed Expectations are created through the GX Cloud UI.
-- API-managed Expectations are created with the API in a GX Cloud Data Context.
+- API-managed Expectations are created through the GX Cloud API.
```
Comment (AkshatAtGX) [inline docs/docusaurus/docs/cloud/expectations/manage_expectations.md]: I think "using" is better than "through", for both UI and API, but no strong opinion.

File: docs/docusaurus/docs/cloud/expectations/manage_expectations.md [hunk 5]
```diff
@@ -236,21 +236,21 @@ If your Data Asset has at least one DATE or DATETIME column, you can define a Ba
...
+| Edit               | [Edit parameters](/cloud/expectations/manage_expectations.md#edit-an-expectation) with the GX Cloud UI                                                                                              | [Edit parameters with the GX Cloud API](/reference/api/expectations/Expectation_class.mdx) or the GX Cloud UI                                                                                                                                                                                                                        |
+| Batch              | [Define a Batch](/cloud/expectations/manage_expectations.md#optional-define-a-batch) in the Cloud UI                                                                                                | Define a Batch with the Cloud API when connecting to [SQL](/core/connect_to_data/sql_data/sql_data.md#create-a-batch-definition), [filesystem](/core/connect_to_data/filesystem_data/filesystem_data.md#create-a-batch-definition), or [dataframe](/core/connect_to_data/dataframes/dataframes.md#create-a-batch-definition) data |
```
Comment (AkshatAtGX) [inline docs/docusaurus/docs/cloud/expectations/manage_expectations.md]: We say "Cloud API" here and "GX Cloud API" in the line above. We should be consistent, imo.

File: docs/docusaurus/docs/cloud/overview/gx_cloud_overview.md [hunk 6]
```diff
@@ -58,14 +58,14 @@ There are a variety of GX Cloud features that support additional enhancements to
...
 
 * **GX Cloud frontend web UI**. Enables you to manage and validate your organization's data quality without running code and provides shared visibility into your organization's Validation Results history.
 
 * **GX Cloud data storage**. Stores the configurations for your organization's Data Sources, Data Assets, Expectations, and Validations alongside your organization's Validation Result histories and Data Asset descriptive metrics.
 
 * **GX Cloud backend application**. Contains the necessary logic and compute to connect to data and run queries. The specifics of how the GX Cloud backend connects to your data is described in [Deployment patterns](/cloud/deploy/deployment_patterns.md).
 
-* **GX Core Python client**. Enables you to interact programmatically with GX Cloud entities and workflows created via the API. The [GX Core Python client](/core/introduction/introduction.mdx) can complement and extend your web UI-created workflows.
\ No newline at end of file
+* **GX Cloud Python API**. Enables you to [interact programmatically](/reference/index.md) with GX Cloud entities and workflows.
```
Comment (AkshatAtGX) [inline docs/docusaurus/docs/cloud/overview/gx_cloud_overview.md]: I recommend using "GX Cloud API". It's fair to clarify that this is python, since using the API uses python, but I'm thinking this is an opportunity to draw a distinction between what it is (the API) and how it is used (python, REST, etc.).

File: docs/docusaurus/docs/gx_welcome.md [hunk 7]
```diff
@@ -30,6 +30,6 @@ import GXCard from '@site/src/components/GXCard';
 
   <LinkCard topIcon label="Learn about GX features" description="Use tutorials and conceptual topics to learn more about GX features and functionality." to="/reference/learn" icon="/img/overview_icon.svg" />
 
-  <LinkCard topIcon label="Reference the GX Core Python API" description="View Python API reference for GX Core classes and methods." to="/reference" icon="/img/api_icon.svg" />
+  <LinkCard topIcon label="Reference the GX API" description="View API reference for details on classes and methods." to="/reference" icon="/img/api_icon.svg" />
```
Comment (AkshatAtGX) [inline docs/docusaurus/docs/gx_welcome.md:33]: Is this meant to be "GX Cloud API"? What is "GX API", if it's not the Cloud API?

**Review by AkshatAtGX — CHANGES_REQUESTED** [review]

File: docs/docusaurus/docs/cloud/validations/manage_validations.md [hunk 8]
```diff
@@ -7,7 +7,7 @@ description: Create and manage Validations in GX Cloud.
 You can manually run a Validation through the GX Cloud UI. This is useful for exploring your data and fine-tuning your Expectations. To run recurring Validations, use a [schedule](/docs/cloud/schedules/manage_schedules.md) or an [orchestrator](/cloud/connect/connect_airflow.md).
 
 :::tip Manual validations are for GX-managed Expectations only
-To run a validation for an [API-managed Expectation](/cloud/expectations/manage_expectations.md#gx-managed-vs-api-managed-expectations), create and run a [Validation Definition](/core/run_validations/create_a_validation_definition.md).
+To run a validation for an [API-managed Expectation](/cloud/expectations/manage_expectations.md#gx-managed-vs-api-managed-expectations), use the GX Cloud UI to generate the necessary code. For the Data Asset of interest, go to the **Validations** tab, click the code snippet icon next to the **Validate** button, and then click **Generate snippet**.
```
Comment (annabarr) [inline docs/docusaurus/docs/cloud/validations/manage_validations.md]: Do you think it's worth calling out that you will need to select the API-managed suite in the left-hand panel?

**Review by annabarr — APPROVED** [review]
I like the diagram update!

File: README.md (hunk 1 above)
Comment (klavavej) [reply README.md:60]: I was just being a bit hand-wavey in disambiguating from "API". If this concept was an "application programming interface" I assumed it's reasonable to call it just an "interface" 😅 see below for other suggestions

File: README.md (hunk 2 above)
Comment (klavavej) [reply README.md:63]: yes, that's correct

Here's another option. What do you think?

```suggestion
🟢 Ready. Have clear and public extension points.
```

File: docs/docusaurus/docs/cloud/validations/manage_validations.md (hunk 8 above)
Comment (klavavej) [reply docs/docusaurus/docs/cloud/validations/manage_validations.md]: oh, yes, I didn't realize that was the case. thanks for this note

File: docs/docusaurus/docs/gx_welcome.md (hunk 7 above)
Comment (klavavej) [reply docs/docusaurus/docs/gx_welcome.md:33]: This is something I intended to revisit as part of future IA changes regarding the API reference. The API reference is still shared between Cloud and Core so I've left the top nav item for it with the less than ideal label "GX API". These cards on the docs landing page all have 1:1 relationships with top nav items so I've used the transitional language on this card as well. While less than ideal, I believe this is still an improvement over the old language  of "GX Core Python API".

Comment (klavavej) [reply docs/docusaurus/docs/gx_welcome.md:33]: Screenshot 2025-06-03 at 11 59 31 AM

**Review by AkshatAtGX — APPROVED** [review]
I think "interface" is the best option of the two. Let's stick with that one for now?
Everything else looks good after the changes. ACK your comment on GX API being addressed as part of broader IA changes.
